feathr_project/feathr/registry/_feature_registry_purview.py

```diff
@@ -1097,9 +1297,11 @@ def _get_feature_type_from_hocon(self, input_str: str) -> FeatureType:
         Returns:
             FeatureType: feature type that can be used by Python
         """
+        if not input_str:
+            return None
         conf = ConfigFactory.parse_string(input_str)
-        valType = conf.get_string('type.valType')
-        dimensionType = conf.get_string('type.dimensionType')
+        valType = conf.get_string('valType') if 'valType' in conf else conf.get_string('type.valType')
```

Comment:
Are these to support deprecated data structure? Shall we have some brief inline comment for better maintenance experience later?

Reply:
Good point, will add comments in later PR for awarness